### ISSUE TITLE
[BUG] GCP Outputting New IK Motion Over Wrong Time Frames

### ISSUE THREAD

**Issue by bfregly** [rcnl-org/nmsm-core#350]
**Describe the bug**
I just tried to use the IK motion output by my GCP run as an initial guess for TO. I first animated the one-cycle motion output by GCP, which used a time window of 2.235 sec to 3.280 sec. That time window was the time vector present in the GCP-generated IK motion file, as expected. However, right foot heelstrike was not occurring at the start of the animation at 2.235 sec.

I then animated the original IK motion file, which runs from 0 to 4 sec. It turns out that 2.235 sec in the GCP-output IK data file corresponds to 0 sec in the original IK motion file. So GCP output the wrong time frames in the original IK data file (0 to 1.045 sec) but with the correct time window (2.235 sec to 3.280).

Load the attached IK motion files into OpenSim to visualize the problem:
Trial12_Gait_IK_results_filtered.mot - Original IK motion file running from 0 to 4 sec.
Trial12_Gait_IK_results_filtered_cycle.sto - Original IK motion file cropped from 2.235 to 3.280 sec with time vector changed to 0 to 1.045 sec.
Trial12_Gait_IK_results_filtered_gcp_cycle.mot - IK motion file output by GCP with time vector running from 2.235 to 3.280 sec. When animated, the motion corresponds to time range 0 to 1.045 sec in Trial12_Gait_IK_results_filtered.mot.
Trial12_Gait_IK_results_filtered_gcp_time_shifted_cycle.mot - IK motion file output by GCP with time vector changed to 0 to 1.045 sec. When animated alongside Trial12_Gait_IK_results_filtered_cycle.sto, the two motions should look almost identical, but they do not.

Bottom line - Somehow GCP is running IK using marker data from the wrong time frames (t = 0 to t = tfinal) and then outputting the IK motion file using the correct time range.

To prepare for my TO run (where I always make time

**To Reproduce**
See above.

**Expected behavior**
See above.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows
 - OpenSim Version: 4.5
 - MATLAB Version: 2023b
 - NMSM Pipeline Version: 1.3.2

**Additional context**
[GCPBug.zip](https://github.com/user-attachments/files/18136842/GCPBug.zip)

**Comment by SpencerTWilliams** [comment]
Could you also upload your input data and GCP settings file?

**Comment by bfregly** [comment]
Here are all the files needed to run GCP.
[GCP.zip](https://github.com/user-attachments/files/18157809/GCP.zip)

**Comment by SpencerTWilliams** [comment]
Fixed on dev and gcp-mex by checking the time range of the intial input motion file
